### ISSUE TITLE
Definition lists break when switching from wysiwyg editor to markdown editor and back

### ISSUE THREAD

**Issue by ddouglas87** [laurent22/joplin#8489]
https://joplinapp.org/markdown/ lists [Deflist](https://github.com/markdown-it/markdown-it-deflist) as a supported feature.  Yet when I type in the definition list syntax nothing happens.

Eg, copy-pasting in the example code

```
Term 1

:   Definition 1

Term 2 with *inline markup*

:   Definition 2

        { some code, part of Definition 2 }

    Third paragraph of definition 2.
```
does nothing.

## Environment

Joplin version: 2.11.11
Platform: Linux
OS specifics: Mint

## Describe what you expected to happen

It should look like [this](https://global.discourse-cdn.com/standard14/uploads/cozic/original/2X/8/872f45ab39c18931f58e3f1d6d7c501e576691b8.png). Source: https://discourse.joplinapp.org/t/definition-lists-rendering/5945

**Comment by Daeraxa** [comment]
I'm not sure I'm entirely following, what are you actually seeing?

The image you linked saying this is what it *should* look like is the same one the topic you listed says it *shouldn't* look like, which, as far as I understand, is incorrect.

When I try with that example I get:

![image](https://github.com/laurent22/joplin/assets/58074586/c28b7b4b-1f03-4536-845a-11df48ad27e7)

which is:

```html
<dl>
    <dt>Term 1</dt>
    <dd>
        <p>Definition 1</p>
    </dd>
    <dt>Term 2 with <em>inline markup</em></dt>
    <dd>
        <p>Definition 2</p>
        <pre><code>  { some code, part of Definition 2 }
        </code></pre>
        <p>Third paragraph of definition 2.</p>
    </dd>
</dl>
```

Edit: Have you definitely turned the feature on in your markdown options?

**Comment by ddouglas87** [comment]
>Edit: Have you definitely turned the feature on in your markdown options?

There is no instruction saying that needs to be done, or where or how to do that.

---

I went into Tools Menu -> Options -> Markdown -> clicked Enable deflist syntax.

It works!!  Thanks! :D

Maybe consider adding a note to https://joplinapp.org/markdown/ that the feature will not work out of the box without being enabled.

**Comment by ddouglas87** [comment]
>Edit: Have you definitely turned the feature on in your markdown options?

Thanks for the quick reply.  

I went into Tools Menu -> Options -> Markdown -> clicked Enable deflist syntax.

Speaking of which, having to enable it isn't mentioned anywhere.  Maybe consider adding a note to https://joplinapp.org/markdown/ that the feature will not work out of the box without being enabled?  I imagine that would help quite a few people.

---

Further information:

I apologize, I don't know what the mode I'm in is called: When I click the Toggle Editors button towards the top right, to the right option (the paper and a pencil icon), it doesn't let you initially write deflists, and it breaks deflists switching modes.

Here's a screenshot of what I'm seeing: https://i.ibb.co/y5CTg3q/deflist-not-working.png  But if I click to change mode and back (the M down arrow button), here's now what it looks like: https://i.ibb.co/r60pSp1/deflist-working-properly.png

If I add more text and change the mode by clicking Toggle Editors and back, deflist breaks.  See pic: https://i.ibb.co/7Nx86GQ/deflist-not-working-again.png  Old deflists collapse if new writing is added and then viewing mode is switched back and forth.

Here's another pic of it: https://i.ibb.co/c1fwVcS/deflist-not-working-diff-view.png  There was 3 deflists there.  The first two collapsed when changing viewing mode.

To summarize, deflist isn't working in the note and pencil viewing mode.  It works when switching to markdown and back, then breaks again when switching to markdown and back.  I'd prefer to just be in the pencil mode, and just write deflists without having to switch back and forth.

**Comment by Daeraxa** [comment]
> Speaking of which, having to enable it isn't mentioned anywhere.  Maybe consider adding a note to https://joplinapp.org/markdown/ that the feature will not work out of the box without being enabled?  I imagine that would help quite a few people.

It is mentioned here - https://joplinapp.org/markdown/#markdown-plugins

> Unlike regular plugins, Markdown plugins must be enabled in the markdown section of the [Configuration screen](https://joplinapp.org/config_screen/). Not all of the plugins are enabled by default, if the "enabled" field is 'no' below, then open the option screen to enable the plugin. Plugins can be disabled in the same manner.

What you are doing isn't supported, the plugin mentions that it isn't supported by the richtext/wysiwyg editor:

![image](https://github.com/laurent22/joplin/assets/58074586/b30d3f1e-b92e-4438-ace8-94f33dcc53b8)

This is only supported for Markdown editing and weird things can happen if it is used outside of that mode.

**Comment by ddouglas87** [comment]
>This is only supported for Markdown editing and weird things can happen if it is used outside of that mode.

So basically, if I use deflists I can never click on the wysiwyg editor, or it will break the document.  When clicking back to the markdown editor it stays broken.  That can't be the intended use case?

(Thanks for dealing with my vagueness above.  I didn't know the full extent of the issue until I played with it a few times.  If you want I can rewrite the top post above to clarify the issue.  Anything to help.)

**Comment by Daeraxa** [comment]
The main issue is that TinyMCE (the richtext editor) doesn't have any deflist capabilities in the sense that I don't think it is in any way possible to create a new deflist.

However editing the deflist doesn't seem to be a problem, the deflist structure seems to be kept in tact so long as you don't actually delete any of its elements which causes it to go a bit weird.

When Joplin converts it back it seems to completely ignore those tags and converts them to `<p>`, if you add the jopNoMd-Conv class to the `d1` element it just converts it to html. Basically I just don't think `turndown` has anything in it to recognise deflists in order to convert it.

So we still have a problem. Sure, in theory deflist could be added to `turndown` in order to properly "back-convert" from the richtext editor but the editor itself is incapable of handling deflists so you could still easily unintentionally break things. This leads to a weird kind of "half support" which I don't really think is wanted either - either it is supported or it isn't.

**Comment by ddouglas87** [comment]
> You may comment on the issue and I will leave it open.

Bug has not been fixed.
